llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2113,3 +2113,18 @@ bool RISCVTTIImpl::shouldConsiderAddressTypePromotion(
   }
   return Considerable;
 }
+
+RISCVTTIImpl::TTI::MemCmpExpansionOptions
+RISCVTTIImpl::enableMemCmpExpansion(bool OptSize, bool IsZeroCmp) const {
+  TTI::MemCmpExpansionOptions Options;
+  // FIXME: Vector haven't been tested.
+  Options.AllowOverlappingLoads =
+      (ST->enableUnalignedScalarMem() || ST->enableUnalignedVectorMem());
+  Options.MaxNumLoads = TLI->getMaxExpandSizeMemcmp(OptSize);
+  Options.NumLoadsPerBlock = Options.MaxNumLoads;
+  if (ST->is64Bit())
+    Options.LoadSizes.push_back(8);
+  llvm::append_range(Options.LoadSizes, ArrayRef({4, 2, 1}));
+  Options.AllowedTailExpansions = {3, 5, 6};
```

Comment:
I checked what this is used for, and it seems like we'll try to emit a 3 byte ld/st?  That seems wrong.

Reply:
Yes, it seems we will generate i40/i48/... loads (see also the tests in https://github.com/llvm/llvm-project/pull/70469).
This may not work for RISC-V.

Reply:
I removed it first. If RISC-V can benefit from it, we can add it back as a follow-up.